Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py

```diff
@@ -54,20 +40,97 @@ def needs_processing(property_value, process_reduction_type):
             if run.hasProperty('ProcessedAs'):
                 process = run.getLogData('ProcessedAs').value
                 if process == process_reduction_type:
-                    logger.notice('Reusing {0} workspace: {1}'.format(process_reduction_type, ws_name))
+                    logger.notice('Reusing {0} workspace: {1}'
+                                  .format(process_reduction_type, ws_name))
                 else:
-                    logger.warning(
-                        '{0} workspace found, but processed differently: {1}'.format(process_reduction_type, ws_name))
+                    logger.warning('{0} workspace found, but processed '
+                                   'differently: {1}'
+                                   .format(process_reduction_type, ws_name))
                     do_process = True
             else:
-                logger.warning(
-                    '{0} workspace found, but missing the ProcessedAs flag: {1}'.format(process_reduction_type, ws_name))
+                logger.warning('{0} workspace found, but missing the '
+                               'ProcessedAs flag: {1}'
+                               .format(process_reduction_type, ws_name))
                 do_process = True
         else:
             do_process = True
     return [do_process, ws_name]
 
 
+def needs_loading(property_value, loading_reduction_type):
+    """
+    Checks whether a given unary input needs loading or is already loaded in
+    ADS.
+    @param property_value: the string value of the corresponding FileProperty
+    @param loading_reduction_type : the reduction_type of input to load
+    """
+    loading = False
+    ws_name = ''
+    if property_value:
+        ws_name = path.splitext(path.basename(property_value))[0]
+        if mtd.doesExist(ws_name):
+            logger.notice('Reusing {0} workspace: {1}'
+                          .format(loading_reduction_type, ws_name))
+        else:
+            loading = True
+    return [loading, ws_name]
+
+
+class AutoProcessContext:
```

Comment:
Shall we put the context definition in a separate file?

Reply:
As discussed, because this class is only used by SANSILLAutoProcess, we decided to keep it in that file.